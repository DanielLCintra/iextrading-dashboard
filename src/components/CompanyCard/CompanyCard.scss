.CompanyCard {
  margin-bottom: 15px;
  position: relative;

  &Link {
    color: inherit;
    text-decoration: none;
    text-decoration-line: none;
    text-decoration-style: none;
    text-decoration-color: inherit;
  }

  &Description {
    margin: 15px 0 !important;
  }

  &Tag {
    margin-right: 5px;
  }

  &BlankIcon {
    position: absolute;
    top: 10px;
    right: 10px;
    color: royalblue;
  }
}
