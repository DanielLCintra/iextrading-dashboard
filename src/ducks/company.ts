import { AnyAction, Dispatch } from 'redux';
import Api from '../utils/Api';

export const DUCK_NAME = 'company';

// Interfaces

export interface ICompany {
  symbol: string,
  companyName: string,
  industry: string,
  website: string,
  description: string,
  CEO: string,
  sector: string,
  tags: string[],
}

export interface ICompanyState {
  data: ICompany | null,
  error: Error | null,
  loading: boolean,
}

// Action types

export const LOAD_COMPANY_STARTED = `${DUCK_NAME}/LOAD_COMPANY_STARTED`;
interface ILoadCompanyStarted {
  symbol: string,
  type: typeof LOAD_COMPANY_STARTED,
}

export const LOAD_COMPANY_SUCCEED = `${DUCK_NAME}/LOAD_COMPANY_SUCCEED`;
interface ILoadCompanySucceed {
  symbol: string,
  type: typeof LOAD_COMPANY_SUCCEED,
  data: ICompany[],
}

export const LOAD_COMPANY_FAILED = `${DUCK_NAME}/LOAD_COMPANY_FAILED`;
interface ILoadCompanyFailed {
  symbol: string,
  type: typeof LOAD_COMPANY_FAILED,
  error: Error,
}

// Action creators

export const loadCompanyStarted = (symbol:string):ILoadCompanyStarted =>
    ({ symbol, type: LOAD_COMPANY_STARTED });

export const loadCompanySucceed = (symbol:string, data:ICompany[]):ILoadCompanySucceed =>
    ({ symbol, type: LOAD_COMPANY_SUCCEED, data });

export const loadCompanyFailed = (symbol:string, error:Error):ILoadCompanyFailed =>
  ({ symbol, type: LOAD_COMPANY_FAILED, error });

// Thunks

export const loadCompany = (symbol:string) =>
  async (dispatch:Dispatch) => {
    dispatch(loadCompanyStarted(symbol));
    try {
      const results = await Api.instance.getCompany(symbol);
      dispatch(loadCompanySucceed(symbol, results.data));
      return Promise.resolve(results);
    } catch (error) {
      dispatch(loadCompanyFailed(symbol, error));
      return Promise.reject(error);
    }
  };

// Reducer

export const INITIAL_STATE:Map<string, ICompanyState> = new Map();

const reducer = (state = INITIAL_STATE, action:AnyAction) => {
  switch (action.type) {
  case LOAD_COMPANY_STARTED:
    return new Map(state)
      .set(action.symbol, {
        data: null,
        error: null,
        loading: true,
      });
  case LOAD_COMPANY_SUCCEED:
    return new Map(state)
    .set(action.symbol, {
      data: action.data,
      error: null,
      loading: false,
    });
  case LOAD_COMPANY_FAILED:
    return new Map(state)
    .set(action.symbol, {
      data: null,
      error: action.error,
      loading: false,
    });;
  default: return state;
  }
};

export default reducer;
