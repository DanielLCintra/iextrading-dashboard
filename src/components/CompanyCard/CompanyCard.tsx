import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Chip from '@material-ui/core/Chip';
import Icon from '@material-ui/core/Icon';
import Typography from '@material-ui/core/Typography';
import * as React from 'react';
import { ICompany } from '../../ducks/company';
import classes from './CompanyCard.scss';

export interface ICompanyCardProps {
  company: ICompany,
};

class CompanyCard extends React.Component<ICompanyCardProps> {
  public render() {
    const company = this.props.company;
    return (
        <Card className={classes.CompanyCard}>
          <CardContent>
            <a
              href={company.website}
              title={`Go to ${company.companyName} website`}
              className={classes.CompanyCardLink}
              target="_blank"
            >
              <Icon>open_in_new</Icon>
            </a>
            <Typography
              className={classes.CompanyCardCompanyName}
              color="textSecondary"
            >
              {company.companyName}
            </Typography>
            <Typography
              className={classes.CompanyCardCode}
              variant="headline"
              component="h2"
            >
              {company.symbol}
            </Typography>
            <Typography
              className={classes.CompanyCardSector}
              color="textSecondary"
            >
              {company.sector}
            </Typography>
            <Typography
              className={classes.CompanyCardCEO}
              color="textSecondary"
            >
              CEO: {company.CEO}
            </Typography>
            <Typography
              className={classes.CompanyCardDescription}
              color="textSecondary"
            >
              {company.description}
            </Typography>
            <div className={classes.CompanyCardTags}>
              {
                company.tags.map((tag:string) => (
                  <Chip
                    label={tag}
                    key={tag}
                    className={classes.CompanyCardTag}
                  />
                ))
              }
            </div>
          </CardContent>
        </Card>
    );
  }
}

export default CompanyCard;
