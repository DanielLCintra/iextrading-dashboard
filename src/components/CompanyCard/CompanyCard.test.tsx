import { mount } from 'enzyme';
import * as React from 'react';
import { createFakeCompany } from '../../ducks/company.test';
import CompanyCard, { ICompanyCardProps } from './CompanyCard';

describe('CompanyCard', () => {
  const props:ICompanyCardProps = {
    company: createFakeCompany('TST'),
  };

  const wrapper = mount(<CompanyCard {...props} />);

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
