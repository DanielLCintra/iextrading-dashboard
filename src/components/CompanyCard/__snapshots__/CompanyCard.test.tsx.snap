// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyCard renders without crashing 1`] = `
<CompanyCard
  company={
    Object {
      "CEO": "Sr. Test",
      "companyName": "Testing Co",
      "description": "Minim do fugiat dolor fugiat excepteur nulla est aliquip.",
      "industry": "Tests",
      "sector": "Tests",
      "symbol": "TST",
      "tags": Array [
        "T",
        "E",
        "S",
        "T",
      ],
      "website": "http://test.com",
    }
  }
>
  <a
    href="http://test.com"
    target="_blank"
    title="Go to Testing Co website"
  >
    <WithStyles(Card)>
      <Card
        classes={
          Object {
            "root": "MuiCard-root-1",
          }
        }
        raised={false}
      >
        <WithStyles(Paper)
          className="MuiCard-root-1"
          elevation={1}
        >
          <Paper
            className="MuiCard-root-1"
            classes={
              Object {
                "elevation0": "MuiPaper-elevation0-4",
                "elevation1": "MuiPaper-elevation1-5",
                "elevation10": "MuiPaper-elevation10-14",
                "elevation11": "MuiPaper-elevation11-15",
                "elevation12": "MuiPaper-elevation12-16",
                "elevation13": "MuiPaper-elevation13-17",
                "elevation14": "MuiPaper-elevation14-18",
                "elevation15": "MuiPaper-elevation15-19",
                "elevation16": "MuiPaper-elevation16-20",
                "elevation17": "MuiPaper-elevation17-21",
                "elevation18": "MuiPaper-elevation18-22",
                "elevation19": "MuiPaper-elevation19-23",
                "elevation2": "MuiPaper-elevation2-6",
                "elevation20": "MuiPaper-elevation20-24",
                "elevation21": "MuiPaper-elevation21-25",
                "elevation22": "MuiPaper-elevation22-26",
                "elevation23": "MuiPaper-elevation23-27",
                "elevation24": "MuiPaper-elevation24-28",
                "elevation3": "MuiPaper-elevation3-7",
                "elevation4": "MuiPaper-elevation4-8",
                "elevation5": "MuiPaper-elevation5-9",
                "elevation6": "MuiPaper-elevation6-10",
                "elevation7": "MuiPaper-elevation7-11",
                "elevation8": "MuiPaper-elevation8-12",
                "elevation9": "MuiPaper-elevation9-13",
                "root": "MuiPaper-root-2",
                "rounded": "MuiPaper-rounded-3",
              }
            }
            component="div"
            elevation={1}
            square={false}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <WithStyles(CardContent)>
                <CardContent
                  classes={
                    Object {
                      "root": "MuiCardContent-root-29",
                    }
                  }
                  component="div"
                >
                  <div
                    className="MuiCardContent-root-29"
                  >
                    <WithStyles(Icon)>
                      <Icon
                        classes={
                          Object {
                            "colorAction": "MuiIcon-colorAction-33",
                            "colorDisabled": "MuiIcon-colorDisabled-35",
                            "colorError": "MuiIcon-colorError-34",
                            "colorPrimary": "MuiIcon-colorPrimary-31",
                            "colorSecondary": "MuiIcon-colorSecondary-32",
                            "fontSizeInherit": "MuiIcon-fontSizeInherit-36",
                            "root": "MuiIcon-root-30",
                          }
                        }
                        color="inherit"
                        fontSize="default"
                      >
                        <span
                          aria-hidden="true"
                          className="material-icons MuiIcon-root-30"
                        >
                          open_in_new
                        </span>
                      </Icon>
                    </WithStyles(Icon)>
                    <WithStyles(Typography)
                      color="textSecondary"
                    >
                      <Typography
                        align="inherit"
                        classes={
                          Object {
                            "alignCenter": "MuiTypography-alignCenter-50",
                            "alignJustify": "MuiTypography-alignJustify-52",
                            "alignLeft": "MuiTypography-alignLeft-49",
                            "alignRight": "MuiTypography-alignRight-51",
                            "body1": "MuiTypography-body1-46",
                            "body2": "MuiTypography-body2-45",
                            "button": "MuiTypography-button-48",
                            "caption": "MuiTypography-caption-47",
                            "colorError": "MuiTypography-colorError-61",
                            "colorInherit": "MuiTypography-colorInherit-56",
                            "colorPrimary": "MuiTypography-colorPrimary-57",
                            "colorSecondary": "MuiTypography-colorSecondary-58",
                            "colorTextPrimary": "MuiTypography-colorTextPrimary-59",
                            "colorTextSecondary": "MuiTypography-colorTextSecondary-60",
                            "display1": "MuiTypography-display1-41",
                            "display2": "MuiTypography-display2-40",
                            "display3": "MuiTypography-display3-39",
                            "display4": "MuiTypography-display4-38",
                            "gutterBottom": "MuiTypography-gutterBottom-54",
                            "headline": "MuiTypography-headline-42",
                            "noWrap": "MuiTypography-noWrap-53",
                            "paragraph": "MuiTypography-paragraph-55",
                            "root": "MuiTypography-root-37",
                            "subheading": "MuiTypography-subheading-44",
                            "title": "MuiTypography-title-43",
                          }
                        }
                        color="textSecondary"
                        gutterBottom={false}
                        headlineMapping={
                          Object {
                            "body1": "p",
                            "body2": "aside",
                            "display1": "h1",
                            "display2": "h1",
                            "display3": "h1",
                            "display4": "h1",
                            "headline": "h1",
                            "subheading": "h3",
                            "title": "h2",
                          }
                        }
                        noWrap={false}
                        paragraph={false}
                        variant="body1"
                      >
                        <p
                          className="MuiTypography-root-37 MuiTypography-body1-46 MuiTypography-colorTextSecondary-60"
                        >
                          Testing Co
                        </p>
                      </Typography>
                    </WithStyles(Typography)>
                    <WithStyles(Typography)
                      component="h2"
                      variant="headline"
                    >
                      <Typography
                        align="inherit"
                        classes={
                          Object {
                            "alignCenter": "MuiTypography-alignCenter-50",
                            "alignJustify": "MuiTypography-alignJustify-52",
                            "alignLeft": "MuiTypography-alignLeft-49",
                            "alignRight": "MuiTypography-alignRight-51",
                            "body1": "MuiTypography-body1-46",
                            "body2": "MuiTypography-body2-45",
                            "button": "MuiTypography-button-48",
                            "caption": "MuiTypography-caption-47",
                            "colorError": "MuiTypography-colorError-61",
                            "colorInherit": "MuiTypography-colorInherit-56",
                            "colorPrimary": "MuiTypography-colorPrimary-57",
                            "colorSecondary": "MuiTypography-colorSecondary-58",
                            "colorTextPrimary": "MuiTypography-colorTextPrimary-59",
                            "colorTextSecondary": "MuiTypography-colorTextSecondary-60",
                            "display1": "MuiTypography-display1-41",
                            "display2": "MuiTypography-display2-40",
                            "display3": "MuiTypography-display3-39",
                            "display4": "MuiTypography-display4-38",
                            "gutterBottom": "MuiTypography-gutterBottom-54",
                            "headline": "MuiTypography-headline-42",
                            "noWrap": "MuiTypography-noWrap-53",
                            "paragraph": "MuiTypography-paragraph-55",
                            "root": "MuiTypography-root-37",
                            "subheading": "MuiTypography-subheading-44",
                            "title": "MuiTypography-title-43",
                          }
                        }
                        color="default"
                        component="h2"
                        gutterBottom={false}
                        headlineMapping={
                          Object {
                            "body1": "p",
                            "body2": "aside",
                            "display1": "h1",
                            "display2": "h1",
                            "display3": "h1",
                            "display4": "h1",
                            "headline": "h1",
                            "subheading": "h3",
                            "title": "h2",
                          }
                        }
                        noWrap={false}
                        paragraph={false}
                        variant="headline"
                      >
                        <h2
                          className="MuiTypography-root-37 MuiTypography-headline-42"
                        >
                          TST
                        </h2>
                      </Typography>
                    </WithStyles(Typography)>
                    <WithStyles(Typography)
                      color="textSecondary"
                    >
                      <Typography
                        align="inherit"
                        classes={
                          Object {
                            "alignCenter": "MuiTypography-alignCenter-50",
                            "alignJustify": "MuiTypography-alignJustify-52",
                            "alignLeft": "MuiTypography-alignLeft-49",
                            "alignRight": "MuiTypography-alignRight-51",
                            "body1": "MuiTypography-body1-46",
                            "body2": "MuiTypography-body2-45",
                            "button": "MuiTypography-button-48",
                            "caption": "MuiTypography-caption-47",
                            "colorError": "MuiTypography-colorError-61",
                            "colorInherit": "MuiTypography-colorInherit-56",
                            "colorPrimary": "MuiTypography-colorPrimary-57",
                            "colorSecondary": "MuiTypography-colorSecondary-58",
                            "colorTextPrimary": "MuiTypography-colorTextPrimary-59",
                            "colorTextSecondary": "MuiTypography-colorTextSecondary-60",
                            "display1": "MuiTypography-display1-41",
                            "display2": "MuiTypography-display2-40",
                            "display3": "MuiTypography-display3-39",
                            "display4": "MuiTypography-display4-38",
                            "gutterBottom": "MuiTypography-gutterBottom-54",
                            "headline": "MuiTypography-headline-42",
                            "noWrap": "MuiTypography-noWrap-53",
                            "paragraph": "MuiTypography-paragraph-55",
                            "root": "MuiTypography-root-37",
                            "subheading": "MuiTypography-subheading-44",
                            "title": "MuiTypography-title-43",
                          }
                        }
                        color="textSecondary"
                        gutterBottom={false}
                        headlineMapping={
                          Object {
                            "body1": "p",
                            "body2": "aside",
                            "display1": "h1",
                            "display2": "h1",
                            "display3": "h1",
                            "display4": "h1",
                            "headline": "h1",
                            "subheading": "h3",
                            "title": "h2",
                          }
                        }
                        noWrap={false}
                        paragraph={false}
                        variant="body1"
                      >
                        <p
                          className="MuiTypography-root-37 MuiTypography-body1-46 MuiTypography-colorTextSecondary-60"
                        >
                          Tests
                        </p>
                      </Typography>
                    </WithStyles(Typography)>
                    <WithStyles(Typography)
                      color="textSecondary"
                    >
                      <Typography
                        align="inherit"
                        classes={
                          Object {
                            "alignCenter": "MuiTypography-alignCenter-50",
                            "alignJustify": "MuiTypography-alignJustify-52",
                            "alignLeft": "MuiTypography-alignLeft-49",
                            "alignRight": "MuiTypography-alignRight-51",
                            "body1": "MuiTypography-body1-46",
                            "body2": "MuiTypography-body2-45",
                            "button": "MuiTypography-button-48",
                            "caption": "MuiTypography-caption-47",
                            "colorError": "MuiTypography-colorError-61",
                            "colorInherit": "MuiTypography-colorInherit-56",
                            "colorPrimary": "MuiTypography-colorPrimary-57",
                            "colorSecondary": "MuiTypography-colorSecondary-58",
                            "colorTextPrimary": "MuiTypography-colorTextPrimary-59",
                            "colorTextSecondary": "MuiTypography-colorTextSecondary-60",
                            "display1": "MuiTypography-display1-41",
                            "display2": "MuiTypography-display2-40",
                            "display3": "MuiTypography-display3-39",
                            "display4": "MuiTypography-display4-38",
                            "gutterBottom": "MuiTypography-gutterBottom-54",
                            "headline": "MuiTypography-headline-42",
                            "noWrap": "MuiTypography-noWrap-53",
                            "paragraph": "MuiTypography-paragraph-55",
                            "root": "MuiTypography-root-37",
                            "subheading": "MuiTypography-subheading-44",
                            "title": "MuiTypography-title-43",
                          }
                        }
                        color="textSecondary"
                        gutterBottom={false}
                        headlineMapping={
                          Object {
                            "body1": "p",
                            "body2": "aside",
                            "display1": "h1",
                            "display2": "h1",
                            "display3": "h1",
                            "display4": "h1",
                            "headline": "h1",
                            "subheading": "h3",
                            "title": "h2",
                          }
                        }
                        noWrap={false}
                        paragraph={false}
                        variant="body1"
                      >
                        <p
                          className="MuiTypography-root-37 MuiTypography-body1-46 MuiTypography-colorTextSecondary-60"
                        >
                          Sr. Test
                        </p>
                      </Typography>
                    </WithStyles(Typography)>
                    <WithStyles(Typography)
                      color="textSecondary"
                    >
                      <Typography
                        align="inherit"
                        classes={
                          Object {
                            "alignCenter": "MuiTypography-alignCenter-50",
                            "alignJustify": "MuiTypography-alignJustify-52",
                            "alignLeft": "MuiTypography-alignLeft-49",
                            "alignRight": "MuiTypography-alignRight-51",
                            "body1": "MuiTypography-body1-46",
                            "body2": "MuiTypography-body2-45",
                            "button": "MuiTypography-button-48",
                            "caption": "MuiTypography-caption-47",
                            "colorError": "MuiTypography-colorError-61",
                            "colorInherit": "MuiTypography-colorInherit-56",
                            "colorPrimary": "MuiTypography-colorPrimary-57",
                            "colorSecondary": "MuiTypography-colorSecondary-58",
                            "colorTextPrimary": "MuiTypography-colorTextPrimary-59",
                            "colorTextSecondary": "MuiTypography-colorTextSecondary-60",
                            "display1": "MuiTypography-display1-41",
                            "display2": "MuiTypography-display2-40",
                            "display3": "MuiTypography-display3-39",
                            "display4": "MuiTypography-display4-38",
                            "gutterBottom": "MuiTypography-gutterBottom-54",
                            "headline": "MuiTypography-headline-42",
                            "noWrap": "MuiTypography-noWrap-53",
                            "paragraph": "MuiTypography-paragraph-55",
                            "root": "MuiTypography-root-37",
                            "subheading": "MuiTypography-subheading-44",
                            "title": "MuiTypography-title-43",
                          }
                        }
                        color="textSecondary"
                        gutterBottom={false}
                        headlineMapping={
                          Object {
                            "body1": "p",
                            "body2": "aside",
                            "display1": "h1",
                            "display2": "h1",
                            "display3": "h1",
                            "display4": "h1",
                            "headline": "h1",
                            "subheading": "h3",
                            "title": "h2",
                          }
                        }
                        noWrap={false}
                        paragraph={false}
                        variant="body1"
                      >
                        <p
                          className="MuiTypography-root-37 MuiTypography-body1-46 MuiTypography-colorTextSecondary-60"
                        >
                          Minim do fugiat dolor fugiat excepteur nulla est aliquip.
                        </p>
                      </Typography>
                    </WithStyles(Typography)>
                    <div>
                      <WithStyles(Chip)
                        key="T"
                        label="T"
                      >
                        <Chip
                          classes={
                            Object {
                              "avatar": "MuiChip-avatar-71",
                              "avatarChildren": "MuiChip-avatarChildren-74",
                              "avatarColorPrimary": "MuiChip-avatarColorPrimary-72",
                              "avatarColorSecondary": "MuiChip-avatarColorSecondary-73",
                              "clickable": "MuiChip-clickable-65",
                              "clickableColorPrimary": "MuiChip-clickableColorPrimary-66",
                              "clickableColorSecondary": "MuiChip-clickableColorSecondary-67",
                              "colorPrimary": "MuiChip-colorPrimary-63",
                              "colorSecondary": "MuiChip-colorSecondary-64",
                              "deletable": "MuiChip-deletable-68",
                              "deletableColorPrimary": "MuiChip-deletableColorPrimary-69",
                              "deletableColorSecondary": "MuiChip-deletableColorSecondary-70",
                              "deleteIcon": "MuiChip-deleteIcon-76",
                              "deleteIconColorPrimary": "MuiChip-deleteIconColorPrimary-77",
                              "deleteIconColorSecondary": "MuiChip-deleteIconColorSecondary-78",
                              "label": "MuiChip-label-75",
                              "root": "MuiChip-root-62",
                            }
                          }
                          clickable={false}
                          color="default"
                          component="div"
                          label="T"
                        >
                          <div
                            className="MuiChip-root-62"
                            onKeyDown={[Function]}
                            role="button"
                            tabIndex={-1}
                          >
                            <span
                              className="MuiChip-label-75"
                            >
                              T
                            </span>
                          </div>
                        </Chip>
                      </WithStyles(Chip)>
                      <WithStyles(Chip)
                        key="E"
                        label="E"
                      >
                        <Chip
                          classes={
                            Object {
                              "avatar": "MuiChip-avatar-71",
                              "avatarChildren": "MuiChip-avatarChildren-74",
                              "avatarColorPrimary": "MuiChip-avatarColorPrimary-72",
                              "avatarColorSecondary": "MuiChip-avatarColorSecondary-73",
                              "clickable": "MuiChip-clickable-65",
                              "clickableColorPrimary": "MuiChip-clickableColorPrimary-66",
                              "clickableColorSecondary": "MuiChip-clickableColorSecondary-67",
                              "colorPrimary": "MuiChip-colorPrimary-63",
                              "colorSecondary": "MuiChip-colorSecondary-64",
                              "deletable": "MuiChip-deletable-68",
                              "deletableColorPrimary": "MuiChip-deletableColorPrimary-69",
                              "deletableColorSecondary": "MuiChip-deletableColorSecondary-70",
                              "deleteIcon": "MuiChip-deleteIcon-76",
                              "deleteIconColorPrimary": "MuiChip-deleteIconColorPrimary-77",
                              "deleteIconColorSecondary": "MuiChip-deleteIconColorSecondary-78",
                              "label": "MuiChip-label-75",
                              "root": "MuiChip-root-62",
                            }
                          }
                          clickable={false}
                          color="default"
                          component="div"
                          label="E"
                        >
                          <div
                            className="MuiChip-root-62"
                            onKeyDown={[Function]}
                            role="button"
                            tabIndex={-1}
                          >
                            <span
                              className="MuiChip-label-75"
                            >
                              E
                            </span>
                          </div>
                        </Chip>
                      </WithStyles(Chip)>
                      <WithStyles(Chip)
                        key="S"
                        label="S"
                      >
                        <Chip
                          classes={
                            Object {
                              "avatar": "MuiChip-avatar-71",
                              "avatarChildren": "MuiChip-avatarChildren-74",
                              "avatarColorPrimary": "MuiChip-avatarColorPrimary-72",
                              "avatarColorSecondary": "MuiChip-avatarColorSecondary-73",
                              "clickable": "MuiChip-clickable-65",
                              "clickableColorPrimary": "MuiChip-clickableColorPrimary-66",
                              "clickableColorSecondary": "MuiChip-clickableColorSecondary-67",
                              "colorPrimary": "MuiChip-colorPrimary-63",
                              "colorSecondary": "MuiChip-colorSecondary-64",
                              "deletable": "MuiChip-deletable-68",
                              "deletableColorPrimary": "MuiChip-deletableColorPrimary-69",
                              "deletableColorSecondary": "MuiChip-deletableColorSecondary-70",
                              "deleteIcon": "MuiChip-deleteIcon-76",
                              "deleteIconColorPrimary": "MuiChip-deleteIconColorPrimary-77",
                              "deleteIconColorSecondary": "MuiChip-deleteIconColorSecondary-78",
                              "label": "MuiChip-label-75",
                              "root": "MuiChip-root-62",
                            }
                          }
                          clickable={false}
                          color="default"
                          component="div"
                          label="S"
                        >
                          <div
                            className="MuiChip-root-62"
                            onKeyDown={[Function]}
                            role="button"
                            tabIndex={-1}
                          >
                            <span
                              className="MuiChip-label-75"
                            >
                              S
                            </span>
                          </div>
                        </Chip>
                      </WithStyles(Chip)>
                      <WithStyles(Chip)
                        key="T"
                        label="T"
                      >
                        <Chip
                          classes={
                            Object {
                              "avatar": "MuiChip-avatar-71",
                              "avatarChildren": "MuiChip-avatarChildren-74",
                              "avatarColorPrimary": "MuiChip-avatarColorPrimary-72",
                              "avatarColorSecondary": "MuiChip-avatarColorSecondary-73",
                              "clickable": "MuiChip-clickable-65",
                              "clickableColorPrimary": "MuiChip-clickableColorPrimary-66",
                              "clickableColorSecondary": "MuiChip-clickableColorSecondary-67",
                              "colorPrimary": "MuiChip-colorPrimary-63",
                              "colorSecondary": "MuiChip-colorSecondary-64",
                              "deletable": "MuiChip-deletable-68",
                              "deletableColorPrimary": "MuiChip-deletableColorPrimary-69",
                              "deletableColorSecondary": "MuiChip-deletableColorSecondary-70",
                              "deleteIcon": "MuiChip-deleteIcon-76",
                              "deleteIconColorPrimary": "MuiChip-deleteIconColorPrimary-77",
                              "deleteIconColorSecondary": "MuiChip-deleteIconColorSecondary-78",
                              "label": "MuiChip-label-75",
                              "root": "MuiChip-root-62",
                            }
                          }
                          clickable={false}
                          color="default"
                          component="div"
                          label="T"
                        >
                          <div
                            className="MuiChip-root-62"
                            onKeyDown={[Function]}
                            role="button"
                            tabIndex={-1}
                          >
                            <span
                              className="MuiChip-label-75"
                            >
                              T
                            </span>
                          </div>
                        </Chip>
                      </WithStyles(Chip)>
                    </div>
                  </div>
                </CardContent>
              </WithStyles(CardContent)>
            </div>
          </Paper>
        </WithStyles(Paper)>
      </Card>
    </WithStyles(Card)>
  </a>
</CompanyCard>
`;
